test/inductor/test_triton_kernels.py

```diff
@@ -3550,6 +3550,279 @@ def f(x, increment):
+        # We need to specify global records to get around an unknown Dynamo bug.
+        # Dynamo appears to not properly update the side effects in the case of an
+        # inline call with a nonlocal value.
+        global records
+        records = {}
+
+        def early_config_prune(configs, named_args, **kwargs):
+            # we need to save the records to the returned config
+            records["run_early_config_prune"] = True
+            if "N" in kwargs and kwargs["N"] == 1024:
+                records["capture_kwargs"] = True
+            if "dst" in named_args and "src" in named_args and len(named_args) == 5:
+                records["capture_named_args"] = True
+            return [configs[0]]
+
+        # this will pick the Config with the largest block size  (128)
+
+        def perf_model(*args, **kwargs):
+            records["run_perf_model"] = True
+            return kwargs["BLOCK_SIZE"] * -1
+
+        if with_perf_model:
+            prune_configs_by = {"perf_model": perf_model, "top_k": 1}
+        else:
+            prune_configs_by = {"early_config_prune": early_config_prune}
+
+        @triton.autotune(
+            configs=[
+                triton.Config(kwargs={"BLOCK_SIZE": 32}),
+                triton.Config(kwargs={"BLOCK_SIZE": 128}),
+            ],
+            key=["N"],
+            prune_configs_by=prune_configs_by,
+        )
+        @triton.jit
+        def prune_by_kernel(
+            dst,
+            src,
+            add_float,
+            string_arg: tl.constexpr,
+            bool_var: tl.constexpr,
+            N,
+            BLOCK_SIZE: tl.constexpr,
+        ):
+            offsets = tl.program_id(0) * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+            x = tl.load(src + offsets, mask=offsets < N)
+            # Let's make sure we always select a block size of 128 based on our perf_model
+            if (string_arg == "TEST" and bool_var) and BLOCK_SIZE == 128:
+                x = x + add_float
+            tl.store(dst + offsets, x, mask=offsets < N)
+
+        torch._dynamo.reset()
+        counter = 0
+
+        # Track how many compiles occur
+        @torch._dynamo.on_compile_start
+        def start_callback():
+            nonlocal counter
+            counter += 1
+
+        @torch.compile(fullgraph=True, backend=backend)
+        def f(dst, src, add_float, string_arg, bool_var, N):
+            grid = lambda META: (triton.cdiv(N, META["BLOCK_SIZE"]),)
+            prune_by_kernel[grid](dst, src, add_float, string_arg, bool_var, N=N)
+
+        N = 1024
+        src = torch.randn(N, device=GPU_TYPE)
+        dst = torch.empty(N, device=GPU_TYPE)
+
+        # first compilation, this prunes the configs
+        f(dst, src, 1.5, "TEST", True, N)
+
+        self.assertEqual(counter, 1)
+
+        # configs has changed (because of the pruning), so this will recompile
+        f(dst, src, 1.5, "TEST", True, N)
+
+        self.assertEqual(counter, 2)
```

Comment:
Why do we recompile here? Nothing has changed on the kernel, configs, callbacks, or arguments.

Reply:
this was happening because of the records dict---fixed now